.github/workflows/inter-branch-merge.yml

```diff
@@ -0,0 +1,62 @@
+name: Usage of Inter-branch merge workflow
+on:
+  push:
+    branches:
+      # TODO - vs16.11
+      # TODO - vs17.0
+      # TODO - vs17.3
+      # TODO - vs17.4
+      # TODO - vs17.6
+      # TODO - vs17.7
+      # TODO - vs17.8
+      # TODO - vs17.9
+      - vs17.10
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  merge-from-to:
+    # strategy here helps with mapping the source and target branches
+    strategy:
+      matrix:
+        source: []
+        target: []
+        include:
+          - source: 'vs16.11'
+            target: 'vs17.0'
+          - source: 'vs17.0'
+            target: 'vs17.3'
+          - source: 'vs17.3'
+            target: 'vs17.4'
+          - source: 'vs17.4'
+            target: 'vs17.6'
+          - source: 'vs17.6'
```

Comment:
note both `vs17.6` and `vs17.7` flow into `vs17.8` even in the original configuration. I hope it's fine and not a typo ?